alf/algorithms/containers.py

```diff
@@ -0,0 +1,622 @@
+                self._loss = Loss()
+
+            def rollout_step(self, inputs, state):
+                return self._step(inputs, state)
+
+            def train_step(self, inputs, state, rollout_info):
+                return self._step(inputs, state)
+
+            def _step(self, inputs, state):
+                predictor_state, delayed = state
+                predicted, predictor_state = self._predictor(inputs, predictor_state)
+                error = (delayed - inputs) ** 2
+                loss_step = self._loss.rollout_step(error)
+                return AlgStep(
+                    output=predicted,
+                    state=(predictor_state, predicted),
+                    info=loss_step.info)
+
+            def calc_loss(info):
+                return self._loss.calc_loss(info)
+
+        alg = PredictAlgorithm(predictor)
+
+    Example 2:
+
+    The following example constructs an actor-critic algorithm:
+
+    .. code-block:: python
+
+        value_net = ValueNetwork(...)
+        actor_net = ActorDistributionNetwork(...)
+
+        alg = SequentialAlg(
+            is_on_policy=True,
+            value=('input.observation', value_net),
+            action_dist=('input.observation', actor_net),
+            action=dist_utils.sample_action_distribution,
+            loss=(ActorCriticInfo(
+                reward='input.reward',
+                step_type='input.step_type',
+                discount='input.discount',
+                action_distribution='action_dist',
+                action='action',
+                value='value'), ActorCriticLoss()),
+            output='action')
+
+    It is equivalent to the following:
+
+    .. code-block:: python
+
+        class ACAlgorithm(Algorithm):
+            def __init__(self, value_net, actor_net):
+                super().__init__(
+                    train_state_spec=(value_net.state_spec, actor_net.state_spec),
+                    is_on_policy=True)
+                self._value_net = value_net
+                self._actor_net = actor_net
+                self._loss = ActorCriticLoss()
+
+            def rollout_step(self, inputs, state):
+                value, value_state = self._value_net(inputs.observation, state[0])
+                action_dist, actor_state = self._actor_net(inputs.observation, state[1])
+                action = dist_utils.sample_action_distribution(action_dist)
+                loss_step = self._loss.rollout_step(ActorCriticInfo(
+                    reward=inputs.reward,
+                    step_type=inputs.step_type,
+                    discount=inputs.discount,
+                    action_distribution=action_dist,
+                    action=action,
+                    value=value))
+                )
+                return AlgStep(
+                    output=action,
+                    state=(value_state, actor_state),
+                    info=loss_step.info)
+
+            def calc_loss(self, info):
```

Comment:
In the two examples, both "calc_loss" is fairly simple: directly call self._loss.calc_loss(info). What if the user wants to customize the calc_loss function and still use this SequentialAlg?